vectorsearch/params/radial_search/add_radial_threshold_to_dataset.py

```diff
@@ -0,0 +1,89 @@
+import h5py
+import numpy as np
+import os
+import sys
+from datetime import datetime
+
+
+def calculate_distances(test_query, train_docs, engine_type, distance_metric='l2_squared'):
+    if distance_metric == 'l2_squared':
+        distances = np.sum((train_docs - test_query) ** 2, axis=1)
+    elif distance_metric == 'cosine':
+        norm_test = np.linalg.norm(test_query)
+        norms_train = np.linalg.norm(train_docs, axis=1)
+        distances = 1 - (np.dot(train_docs, test_query) / (norms_train * norm_test))
+    elif distance_metric == 'inner_product':
+        if engine_type == 'faiss':
+            distances = -np.dot(train_docs, test_query)
+        elif engine_type == 'lucene':
+            distances = np.dot(train_docs, test_query)
+    else:
+        raise ValueError("Unsupported distance metric")
+    return distances
+
+
+def calculate_scores(test_queries, train_docs, distance_metric='l2_squared'):
+    distances = calculate_distances(test_queries, train_docs, distance_metric)
+    if distance_metric == 'l2_squared':
+        scores = 1 / (1 + distances)
+    elif distance_metric == 'cosine':
+        scores = (2 - distances) / 2
+    elif distance_metric == 'inner_product':
+        if engine_type == 'faiss':
+            scores = np.where(distances >= 0, 1 / (1 + distances), 1 - distances)
+        elif engine_type == 'lucene':
+            scores = np.where(distances > 0, distances + 1, 1 / (1 - distances))
+        else:
+            raise ValueError("Unsupported engine type for inner_product")
```

Comment:
nit: can you include the invalid value as part of error message?

Reply:
ack